pkg/machinepool/validation_test.go

```diff
@@ -120,4 +126,103 @@ var _ = Describe("MachinePool validation", func() {
+			Expect(err).To(BeNil())
+		})
+
+		It("Fails when replicas exceed available capacity", func() {
+			capacityReservationID := "cr-12345678"
+			// Mock returns 5 total, 1 available
+			mockClient.EXPECT().GetCapacityReservationDetails(capacityReservationID).
+				Return(int32(5), int32(1), nil)
+
+			err := validateCapacityReservationReplicas(capacityReservationID, 3, mockClient, false, 0, 0)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("cannot set replicas to 3: capacity reservation 'cr-12345678' only has 1 available instance(s)"))
+		})
+
+		It("Passes when autoscaling min replicas are within capacity", func() {
+			capacityReservationID := "cr-12345678"
+			// Mock returns 10 total, 8 available
+			mockClient.EXPECT().GetCapacityReservationDetails(capacityReservationID).
+				Return(int32(10), int32(8), nil)
+
+			err := validateCapacityReservationReplicas(capacityReservationID, 0, mockClient, true, 2, 5)
+			Expect(err).To(BeNil())
+		})
+
+		It("Fails when autoscaling min replicas exceed available capacity", func() {
+			capacityReservationID := "cr-12345678"
+			// Mock returns 10 total, 2 available
+			mockClient.EXPECT().GetCapacityReservationDetails(capacityReservationID).
+				Return(int32(10), int32(2), nil)
+
+			err := validateCapacityReservationReplicas(capacityReservationID, 0, mockClient, true, 3, 5)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("cannot set min replicas to 3: capacity reservation 'cr-12345678' only has 2 available instance(s)"))
+		})
+
+		It("Fails when autoscaling max replicas exceed available capacity", func() {
+			capacityReservationID := "cr-12345678"
+			// Mock returns 10 total, 4 available
+			mockClient.EXPECT().GetCapacityReservationDetails(capacityReservationID).
+				Return(int32(10), int32(4), nil)
+
+			err := validateCapacityReservationReplicas(capacityReservationID, 0, mockClient, true, 1, 5)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("cannot set max replicas to 5: capacity reservation 'cr-12345678' only has 4 available instance(s)"))
+		})
+
+		It("Handles AWS API errors gracefully", func() {
+			capacityReservationID := "cr-12345678"
+			awsError := fmt.Errorf("AWS API error: capacity reservation not found")
+
+			mockClient.EXPECT().GetCapacityReservationDetails(capacityReservationID).
+				Return(int32(0), int32(0), awsError)
+
+			err := validateCapacityReservationReplicas(capacityReservationID, 2, mockClient, false, 0, 0)
+			Expect(err).To(HaveOccurred())
+			// The error is wrapped with additional context
+			Expect(err.Error()).To(ContainSubstring("unable to validate capacity reservation"))
+			Expect(err.Error()).To(ContainSubstring(capacityReservationID))
+			Expect(err.Error()).To(ContainSubstring("AWS API error"))
+		})
+
+		It("Passes when capacity reservation has zero available instances and replicas is zero", func() {
+			capacityReservationID := "cr-12345678"
+			// Mock returns 5 total, 0 available
```

Comment:
NP: These repeated comments stating what the hardcoded values are don't tell me anything. I would suggest saving the hardcoded values (such as `5` and `0`) as variables. Then passing them into the mock expect

this way devs can infer what the tests are doing by just reading the code, instead of having many repeated comments